src/states/elder/mod.rs

```diff
@@ -616,6 +639,25 @@ impl Elder {
         Ok(())
     }
 
+    fn add_to_proving_section_cache(
+        &mut self,
+        proving_secs: Vec<ProvingSection>,
+        sec_info: SectionInfo,
+    ) {
+        if self.chain.is_new_neighbour(&sec_info) {
+            self.set_of_proving_sections
+                .push_back((proving_secs, sec_info));
+        }
+        while self.set_of_proving_sections.len() > PROVING_SECTION_CACHE_SIZE {
```

Comment:
guess if lru cache, this manual stuff may not be needed and also becomes a time bound queue

Reply:
Do we want it to be timebound though?

Reply:
you can choose for either with lru if I remember right or both. I think it does have a max_capacity ctor too